packages/subgraph-service/contracts/SubgraphService.sol

```diff
@@ -309,6 +309,18 @@ contract SubgraphService is
         _closeAllocation(allocationId);
     }
 
+    /**
+     * @notice See {ISubgraphService.closeOverAllocatedAllocation}
+     */
+    function closeOverAllocatedAllocation(address allocationId) external override {
+        Allocation.State memory allocation = allocations.get(allocationId);
+        require(
```

Comment:
We also need to check if the allocation is altruistic (tokens = 0). You shouldn't be able to close altruistic allos even if they are overallocated.

Reply:
I thought about this but how can an altruistic allocation be over-allocated? If tokens == 0 then even if you have 0 available tokens you won't be over allocated. Am I missing something?

Reply:
Right, yeah, makes sense, thank you.